arrows/vxl/image_io.cxx

```diff
@@ -138,6 +143,111 @@ convert_image_helper(const vil_image_view<bool>& src,
   dest = src;
 }
 
+std::string
+plane_filename(const std::string filename, unsigned p)
+{
+  std::string parent_directory =
+    kwiversys::SystemTools::GetParentDirectory(filename);
+  std::string file_name_with_ext =
+    kwiversys::SystemTools::GetFilenameName(filename);
+
+  std::string file_name_no_ext =
+    kwiversys::SystemTools::GetFilenameWithoutLastExtension( file_name_with_ext );
+  std::string file_extension =
+    kwiversys::SystemTools::GetFilenameLastExtension( file_name_with_ext );
+
+  std::vector<std::string> full_path;
+  std::string plane_id = ( p > 0 ? "_" + std::to_string(p) : "" );
+  full_path.push_back("");
+  full_path.push_back(parent_directory);
+  full_path.push_back(file_name_no_ext + plane_id + file_extension);
+  return kwiversys::SystemTools::JoinPath(full_path);
+}
+
+template <typename inP>
+void
+save_image(const vil_image_view<inP>& src,
+           std::string const& filename,
+           bool split_planes=false)
+{
+  if(!split_planes || src.nplanes() == 1)
+  {
+    vil_save(src,filename.c_str());
+  }
+  else
+  {
+    for(unsigned i = 0; i < src.nplanes(); ++i)
+    {
+      vil_save(vil_plane(src,i), plane_filename(filename,i).c_str());
+    }
+  }
+}
+
+// Create a list of filenames representing the non-initial plane files
+std::vector< std::string >
+construct_plane_filenames( std::string const& filename )
+{
+  std::vector< std::string > plane_files;
+
+  for( size_t p = 1;; p++ )
+  {
+    std::string plane_file = plane_filename( filename, p );
+
+    if( ST::FileExists( plane_file ) )
+    {
+      plane_files.push_back( plane_file );
+    }
+    else
+    {
+      break;
+    }
+  }
+  return plane_files;
+}
+
+// Helper function to load images when they are saved out in above format
+template< typename Type >
+vil_image_view< Type >
+load_external_planes( std::string const& filename,
+                      vil_image_view< Type >& first_plane)
+{
+  std::vector< vil_image_view< Type > > images( 1, first_plane );
+
+  size_t p = 1;
+  size_t total_p = first_plane.nplanes();
+
+  auto const plane_filenames = construct_plane_filenames( filename );
+
+  for( auto const& plane_file : plane_filenames )
+  {
+    vil_image_view< Type > plane = vil_load( plane_file.c_str() );
+
+    if( plane.ni() != first_plane.ni() || plane.nj() != first_plane.nj() )
+    {
+      VITAL_THROW( vital::image_type_mismatch_exception, "Input channel size difference" );
+    }
+
+    images.push_back( plane );
+    total_p += plane.nplanes();
+    p++;
```

Comment:
pedantic: it's usually nice to use prefix-increment unless you really need postfix (multiple instances)